### ISSUE TITLE
Dh as a python library: h2o package running problem with deephaven_server.

### ISSUE THREAD

**Issue by Jeremiahcheng1206** [deephaven/deephaven-core#2723]
**Description**
I was trying to running h2o (using GPU to do computation) package with deephaven as a python library.  It gave me this error 'io.deephaven.integrations.python.PythonLogAdapter' object has no attribute 'isatty'. If I run it separately, meaning without starting the deephaven server. It works. Below are the code and output. 

**Steps to reproduce**
### With deephaven server:
```python
from deephaven_server import Server
s = Server(port=10000, jvm_args=["-Xmx4g"])
s.start()
from deephaven import ugp
ugp.auto_locking = True
import h2o
from h2o.automl import H2OAutoML
h2o.init()
data_frame=h2o.import_file("data07292022112829.parquet")
```
### The error:
'io.deephaven.integrations.python.PythonLogAdapter' object has no attribute 'isatty'

### Without deephaven_server:
```python
import h2o
from h2o.automl import H2OAutoML
h2o.init()
data_frame=h2o.import_file("data07292022112829.parquet")
print(data_frame)
```
### The output:
![image](https://user-images.githubusercontent.com/99699304/183514809-63221cf3-6f78-4b16-92a8-dc173deae25e.png)

**Versions**
 - Deephaven: 0.15.0 Dh as a python library
 - OS:  WSL
 - Browser: Google chrome
 - Docker: no docker

**Comment by niloc132** [comment]
Arguably, h2o is misbehaving here, but its a reasonable thing for us to work with. From `h2o/utils/progressbar.py`:
```
        # Fix for PUBDEV-5048. H2O depends on isatty attribute, but some Python Notebooks override stdout and doesn't
        # specify it. The same holds for the encoding attribute bellow
        if not hasattr(sys.stdout, "isatty"):
            sys.stdout.isatty = lambda: False

        if not hasattr(sys.stdout, "encoding"):
            sys.stdout.encoding = sys.getdefaultencoding()

        if file_mode is None: file_mode = not sys.stdout.isatty()
```
Our Java PythonLogAdapter doesn't like being monkey patched like this.

This is broken in both directions: 
 * When starting from python, our PythonLogAdapter needs to support any possible attr that might be set on it - rather than actually supporting everything, it probably makes more sense to handle arbitrary attributes, and thus permit our logger to be monkey-patched like this, rather than imagine we can implement all possible use cases directly.
 * When starting from java and creating python, jpy takes over sys.stdout and sys.stderr, by replacing them with a single instance that funnels all content to just stdout, and leaves behind only the write and flush methods, ignoring the remainder (`writable()`, `writelines()`, `isatty()`, etc). Once running, deephaven further wraps with a Java wrapper around the C wrapper around the underlying native python object. If the point above requires wrapping the underlying stdout/stderr python objects, we may need to go another step here and update jpy to also support other underlying methods.

Note that it might not make sense to actually implement these other methods - java calling python should conform to java's view of the world (which is very limited - java user code has no access to the fd number of either stderr or stdout, nor if either is a tty or not), or should attempt to hook in to python's expectations here. Calling from python into java is likely simpler (and also more important in general to get right), since we can start by just wrapping the python objects, and continue from there.

**Comment by niloc132** [comment]
Use cases to test:
 * from DHaaL, use both standard stderr/stdout, ensure both report isatty as true
 * from DHaaL, redirect one or both stderr/stdout, ensure each reports isatty as expected
 * from DHaaL, confirm the original use case behaves as expected, other properties can be monkey patched
 * from jetty/netty server with a python session, confirm that the original use case behaves as expected

We need to decide if we want to support an encoding other utf-8, since we assume that in many other places.

**Comment by devinrsmith** [comment]
I've just tested this against our main branch - our 0.17.0 release should fix this error.

Fixed as part of #2918, with ultimately the https://github.com/jpy-consortium/jpy/pull/85 being necessary.

I'll close this issue out once we can verify it's been fixed post-release.

**Comment by devinrsmith** [comment]
Tested against our `deephaven-server==0.17.0` release, this seems to be fixed - let us know if you are still seeing issues @Jeremiahcheng1206
